### PR TITLE
[FIX] Fix tests failing due to deprecations in pre-release dependencies

### LINKED ISSUE #3745: Failure in pre-release dependencies CI job
I'm looking into this failure https://github.com/nilearn/nilearn/actions/runs/5186381231/jobs/9347413782
There have been pre-releases for numpy (1.25.0rc1) and scipy (1.11.0rc1) since last week which is around the time this failure seems to have started

### PR DESCRIPTION
**Pull request by ymzayek**
Closes #3745.

Fix test_resampling.py::test_resampling_nan
Fix test_html_report.py::test_nifti_maps_masker_report_list_and_arrays_maps_number

### CODE REVIEW

**Comment by ymzayek** [comment]
I removed testing for the warning in test_resampling_nan because it is tested in the test that follows. But now it fails with another error but only on the prerelease job so maybe it is another thing associated with the numpy upgrade

File: nilearn/maskers/nifti_maps_masker.py [hunk 1]
```diff
@@ -229,7 +229,11 @@ def generate_report(self, displayed_maps=10):
         from nilearn.reporting.html_report import generate_report
 
         if (
-            displayed_maps != "all"
+            (isinstance(displayed_maps, str) and displayed_maps != "all")
+            or (
+                not isinstance(displayed_maps, str)
+                and np.all(displayed_maps != "all")
+            )
```
Comment (bthirion) [inline nilearn/maskers/nifti_maps_masker.py:236]: The logic here is abit complex. Cna you give plain word explanation of the intended behevior

**Comment by ymzayek** [comment]
Failure is unrelated. This should be ready to merge

File: nilearn/maskers/nifti_maps_masker.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/maskers/nifti_maps_masker.py]: Definitely clearer. To me at least.

**Review by Remi-Gau — APPROVED** [review]
This looks good to me.
Thanks @ymzayek

**Review by bthirion — APPROVED** [review]
LGTM, thx

**Comment by bthirion** [comment]
Do you understand the Azure failure ?

**Comment by ymzayek** [comment]
Ok thanks will merge

> Do you understand the Azure failure ?

It's just the timeout issue (being addressed in #3747)

**Pull request merged**